### PR TITLE
Footsteps sounds for bare feet and other mobs

### PR DESCRIPTION
**Pull request by nero1024**
Bare feet and other mobs will now make the correct footstep sounds and added new samples to go with them. Added some helpers to is_helpers.dm, mostly for (future) footstep support but maybe someone else can make use of them.

:cl: Nero1024
soundadd: Bare feet will now make the correct footstep sounds.
soundadd: Other mobs will make the correct footstep sounds.
/:cl:

As others indicated in the very first footsteps PR other users wanted footstep sounds for barefeet. I also noticed that do_footstep is in a bunch of other mobs, but Ian and Xenomorphs don't wear shoes so I fixed that too.

Demos:
Barefeet in action: https://my.mixtape.moe/lrtohp.webm
Xenomrphs clacking around: https://my.mixtape.moe/edzqpc.webm

TODO:
- [x] Fix: Make dead mobs not play footsteep sounds.
- [x] Tweak: Adjust footstep volume for xenos? They're quiet, but perhaps too quiet IMO.

### CODE REVIEW

**Comment by Qustinnus** [comment]
code seems snowflakey

**Comment by nero1024** [comment]
I'll admit that `ishumansimple()` is an absolute mess. Any tips on how It can be tidied up and made presentable?

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -116,22 +116,45 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
...
+#define isasteroidmob(A) (istype (A, /mob/living/simple_animal/asteroid/))
+
 #define ismegafauna(A) (istype(A, /mob/living/simple_animal/hostile/megafauna))
 
 #define isclown(A) (istype(A, /mob/living/simple_animal/hostile/retaliate/clown))
 
+#define isgiantspider(A) (istype(A, /mob/living/simple_animal/hostile/poison/giant_spider))
+
+//Human(oid) simple mobs (it's ugly, barely readable, stupidly long, and can't into newlines)
+#define ishumansimple(A) (istype(A, /mob/living/simple_animal/hostile/cat_butcherer) || istype(A, /mob/living/simple_animal/hostile/faithless) || istype(A, /mob/living/simple_animal/hostile/nanotrasen) || istype(A, /mob/living/simple_animal/hostile/pirate) || istype(A, /mob/living/simple_animal/hostile/russian) || istype(A, /mob/living/simple_animal/hostile/syndicate) || istype(A, /mob/living/simple_animal/hostile/wizard) || istype(A, /mob/living/simple_animal/hostile/zombie) || istype(A, /mob/living/simple_animal/hostile/retaliate/clown) || istype(A, /mob/living/simple_animal/hostile/retaliate/spaceman) || istype(A, /mob/living/simple_animal/hostile/retaliate/nanotrasenpeace))
```
[the remark this answers is not in the document]
Comment (ShizCalev) [reply code/__DEFINES/is_helpers.dm]: atleast make it a typecache or something
https://github.com/tgstation/tgstation/blob/151c4e9b183bf233fa9755ecd0d0bb4d7896e208/code/__DEFINES/is_helpers.dm#L16-L23

File: code/datums/components/footstep.dm [hunk 2]
```diff
@@ -27,13 +29,76 @@
...
+	//for barefooted non-clawed mobs like monkeys
+	if(ismonkey(LM) || isgorilla(LM) || ismook(LM))
+		playsound(T, pick(GLOB.barefootstep[T.barefootstep][1]),
+			GLOB.barefootstep[T.barefootstep][2] * v,
+			TRUE,
+			GLOB.barefootstep[T.barefootstep][3] + e)
+		return
+	
+	//for xenomorphs, dogs, and other clawed mobs
+	if(isalienadult(LM) || isdog(LM) || isfox(LM) || isbear(LM) || ismegaarachnid(LM))
```
Comment (ShizCalev) [inline code/datums/components/footstep.dm]: typecache

**Comment by Bawhoppen** [comment]
I like adding new footstep sounds, but I'm not so sure about the ones for xenos. While I'm not sure how they *should* sound, doesn't seem right to me.

**Comment by nicbn** [comment]
Have an associative list including sounds for barefoot, shoes, claws etc. as a var for turfs. Use defines as keys.
Have a var for mobs and shoes that equals to the define of the sound of such mob or shoe.

When the mob steps, check for shoes. If none or if the var equals zero, use the mob define. Else, use the shoes define.
Then, access the turf list using such define as key.

No istypes, just oop.

**Comment by Qustinnus** [comment]
why not just make the sound lists defines as we do with things such as ambience

**Comment by epochayur** [comment]
ignoring the fact that xeno footsteps are a thinly-veiled excuse to include step sounds for lizards, isn't the entire point of xenos that they're genetically perfect silent killers? it doesn't make sense to me that you could hear them coming

**Comment by Poojawa** [comment]
Xenos have very noisy breathing problems, also not spamming *deathgasp is a terrible Xeno

File: code/__DEFINES/is_helpers.dm [hunk 3]
```diff
@@ -116,22 +116,96 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 
 #define iscat(A) (istype(A, /mob/living/simple_animal/pet/cat))
 
+#define isdog(A) (istype(A, /mob/living/simple_animal/pet/dog))
```
Comment (nicbn) [inline code/__DEFINES/is_helpers.dm]: Remove the unused defines you added

File: code/__DEFINES/is_helpers.dm [hunk 4]
```diff
@@ -116,22 +116,96 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
...
 
+#define isasteroidmob(A) (istype (A, /mob/living/simple_animal/asteroid/))
+
 #define ismegafauna(A) (istype(A, /mob/living/simple_animal/hostile/megafauna))
 
 #define isclown(A) (istype(A, /mob/living/simple_animal/hostile/retaliate/clown))
 
+#define isgiantspider(A) (istype(A, /mob/living/simple_animal/hostile/poison/giant_spider))
+
+GLOBAL_LIST_INIT(simplehuman, typecacheof(list(
```
Comment (nicbn) [inline code/__DEFINES/is_helpers.dm]: You should use a var in these types instead of making a list.

**Review by nicbn — CHANGES_REQUESTED** [review]

**Comment by Militaires** [comment]
footsteps for xenos blows their cover early on and ruins the earlygame stealth you need to establish a colony, do u even xenos bro

**Comment by nero1024** [comment]
I've tried quieting xeno footsteps down, It seems like you have to be right next to them to hear them now. 
Here's a webm:
https://my.mixtape.moe/vwgqlw.webm

**Comment by GuyonBroadway** [comment]
Personally I think the xenos sounds are a little too "slappy" for lack of a better term.

Needs a harder "clack" if you can do such a thing.

**Comment by nero1024** [comment]
Those are the clack-iest claw sounds that I can source.

**Comment by nero1024** [comment]
`if(!T.footstep || LM.buckled || !CHECK_MULTIPLE_BITFIELDS(LM.mobility_flags, MOBILITY_STAND | MOBILITY_MOVE) || LM.throwing || LM.movement_type & (VENTCRAWLING | FLYING))`, specifically mobility_flags, is supposed to check if the mob cannot stand or move correct? How come I'm still hearing footstep sounds from mobs when they're crawling, dead, or lying and being dragged?

**Comment by nero1024** [comment]
I've decided to keep the footstep sounds while crawling, and have the barefeet sounds play as well when crawling (cuz you're using your hands to crawl).

**Comment by Militaires** [comment]
ok what if you're wearing gloves

**Comment by nero1024** [comment]
It would sound about the same.

**Pull request merged**